### DISCUSSION TITLE
Allow a What3Words link to offer OrganicMaps as one of the 'Navigate' options

### DISCUSSION THREAD

**Discussion by PhilStanford** [organicmaps/organicmaps#5996] [Ideas]
I have both Maps.me and OrganicMaps installed on my iPhone. 

If I click on a What3Words hyperlink (eg as shared in an email) then I get taken to the What3Words app on the phone, showing the location, together with a 'Navigate' button.    
If I click that button, I get offered a choice of navigating to that point using a number of mapping apps which are on my phone, including Google Maps, Apple Maps, Citymapper and Maps.me, but not OrganicMaps. 
I am guessing that there must be some API exposed by Maps.me and the other apps, but not by OrganicMaps, which allows the W3W app to 'see' the various navigation apps and transfer coordinates to them (presumably in some standardised, non-proprietary format). 

It would be really useful if OrganicMaps could add this feature - whatever the pros and cons are of using their proprietary algorithm, the W3W codes are widely used. 

NB this is not the same request as was inferred in these ideas - 
#5249 
#2132

**Comment by biodranik** [comment]
You need to contact What3Words app so their developers can add Open in Organic Maps button. Our API is documented here: https://omaps.app/api

**Reply by PhilStanford** [reply]
Hmm I did wonder if that might be the way it needed to work.  I was rather hoping that navigation apps might be able to advertise their services via IOS, and tools like W3W might be able to discover what was installed, rather than W3W needing to maintain a list of possible candidates to check.
I can imagine that they may well be reluctant to extend that list too much but if I can find a means of asking them I'll do so as it would be a useful addition. 
Even without that, Organic Maps is great, thanks for all the hard work.

**Comment by PhilStanford** [comment]
So I contacted What3Words support regarding whether they could allow Organic Maps to be shown in the list of navigation maps offered by the What3Words app and this was their reply: 

> 
> Hi Phil,
> 
> Thanks for getting in touch and for your request.
> 
> Pressing Navigate on what3words displays a list of external apps where the user can start navigation. Unfortunately we don't have control over which apps are available because the developer of each navigation app has to allow what3words to access it.
> 
> What you can do is contact Organic Maps and request they allow what3words to connect to it through the Navigate button. Our developers can help them if they need any advice doing this.
> 

So it does sound as if this functionality would need to be enabled in the Organic Maps app and not in the What3Words app. 
As it is already enabled in Maps.Me my original request remains - could implementing this feature be added to your development request list - I'm sure it would be welcomed and enable more people to move away from Maps.me.   I note that the support email does offer help from the What3Words developers to help with the necessary advice - their address is support@what3words.com and they were quite prompt in replying to my question.

Cheers
Phil

**Reply by biodranik** [reply]
If you have already contacted them, can you please ask them for instructions, what exactly should we add into Organic Maps so their app will detect it and show Organic Maps as a navigation destination?

**Reply by PhilStanford** [reply]
Hi, I will contact them again, but I am probably not best placed to be an interface between two sets of developers!  I will ask them if there is any documentation, or if there is a more specific support address.  But direct contact between someone who understands how Organic Maps works, and the W3W devs, will probably be needed. 
I will report back anyway!

**Comment by PhilStanford** [comment]
Hi, I have had these suggestions back from What3Words support regarding making Organic Maps 'discoverable' as a navigation app from within What3Words:

> Hi Phil,

> This is me again Jana from what3words customer support. I'm pleased to say our developers have shared the information you can pass to Organic Maps.
> 
> For their Android app they'll need to make sure they support default geo call. Here is the official link for android : https://developer.android.com/guide/components/intents-common#ViewMap
> 
> They also need to consider the attached note from Wikipedia: https://en.wikipedia.org/wiki/Geo_URI_scheme#Unofficial_extensions. The most important thing is that the coordinates must be placed as a query parameter and send pair after geo: tag.
> 
> For the iOS app they need to build support for a ‘URL/URI scheme’ - their developers would create this if they don't have it already. The app would need to let us know the details of their URL scheme, and we would need to add to our app which needs to be constructed in a way that accepted a what3words address as a lat/long. It could look something like this:
> 
> *exampleapp://search/w3w?words=square.beats.doormat&lat=35.65574077148753&lon=139.7458314346796*
> I hope this helps Organic Maps add what3words support; if they want to get in touch directly please send them over and we'll be happy to help.
> 
> 

> 

So it does seem as if a combination of actions by both yourselves and What3Words is needed to make this work for the IOS app at least. 

Please do contact them on support@what3words.com for any further information - they do seem very helpful and quick to respond.

**Reply by biodranik** [reply]
Awesome, thank you! It would be even better if you send w3w developers a link to our API that already accepts coordinates to show a point on the map via om:// URL scheme, it's a link that I've shared above: https://omaps.app/api

Let's track this integration for both iOS and Android in the issue. Can you please help to create it?

**Comment by PhilStanford** [comment]
Thanks for your reply - so if I 'Create issue from discussion', I presume I can then mail support@what3words.com  giving them the link to your APIs, and pointing them at a link to the issue.  
Can you then as the developer(s) of OM raise any questions with the W3W supporters directly, and give them the info they need to integrate OM into their app?    I am not sure I can contribute anything more useful just passing messages between your developers and theirs.  But I can kick things off as above if that's helpful.   In the end your mailing their support team direct is probably going to be most efficient, but I understand the point of creating an issue to track things.  
Thanks again - it would be great to get this functionality working for both IOS and Android users.

**Comment by ondohotola** [comment]
I have replaced the Maps.me with the Organic Maps app and, while What3Words still offers Maps.me as the app to navigate with, if I select it, Organic Maps come sup.

While I'd like the button to be the Organic Maps one for aesthetics, this works for me.

**Reply by biodranik** [reply]
Maybe ask them to add Open with Organic Maps button?
